.github/workflows/build.yaml

```diff
@@ -0,0 +1,70 @@
+name: Build
+
+on:
+  workflow_dispatch:
+    inputs:
+      arch:
+        options:
+          - amd64
+          - arm64
+        description: Architecture to build Envoy for Linux
+        type: choice
+        required: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 180
+    defaults:
+      run:
+        working-directory: terraform
+    steps:
+      - uses: actions/checkout@v3
+      # TODO use aws ssm
+      - name: Generate SSH key
+        run: ssh-keygen -t ed25519 -q -N "" -f ~/.ssh/id_ed25519
+      - uses: hashicorp/setup-terraform@v2.0.3
+      - run: terraform init
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/envoy-ci
+          role-duration-seconds: 10800
+          aws-region: us-east-2
+      - name: Set terraform variables
+        run: |
+          cat <<EOF > terraform.tfvars
+          arch = "${{ github.event.inputs.arch }}"
+          public_key_path = "~/.ssh/id_ed25519.pub"
+          EOF
+      - run: terraform apply -auto-approve
+      - id: public-ip
+        # setup-terraform saves stdout of terraform calls to outputs
+        run: terraform output --raw public_ip
+      - name: Wait for VM to be ready
+        timeout-minutes: 2
+        run: |
+          while ! ssh -o StrictHostKeyChecking=no admin@${{ steps.public-ip.outputs.stdout }} 'test -e ready'
+          do
+            sleep 5
+          done
+      - name: Build Envoy
+        run: |
+          ssh -o StrictHostKeyChecking=no admin@${{ steps.public-ip.outputs.stdout }} <<'EOF'
+            git clone https://github.com/kumahq/kuma && cd kuma
+            BUILD_ENVOY_FROM_SOURCES=true make build/envoy
```

Comment:
Can we just have these files here and use these? So we can get rid of this code in Kuma in the future?

Reply:
Yes, definitely. My next steps are PR for darwin and then moving the build stuff to this repo.